cpp/leaky_core.cc

```diff
@@ -179,12 +181,64 @@ bool InputAvailable(int fd) {
   NotImplemented();
 }
 
+SignalHandler::SignalHandler()
+    : signal_queue(), last_sig_num(0), sigwinch_num(UNTRAPPED_SIGWINCH) {
+}
+
+void SignalHandler::EnqueueSignal(int sig_num) {
+  assert(signal_queue != nullptr);
+  signal_queue->append(sig_num);
```

Comment:
We should have an assert() that's it's less than the kMaxSignals, so we don't allocate in the handler


(We haven't really worked out if asserts should be on in release mode; currently they are.)

Reply:
Good catch! This must have gotten lost in the shuffle